### PR TITLE
Hints: add a default hint prompt attempts threshold

### PR DESCRIPTION
**Pull request by Erin007**
Continuation of #36488 
[STAR-1205](https://codedotorg.atlassian.net/browse/STAR-1205)

While we determine how/if we're going to invest in setting the hint prompt attempts threshold based on the logic for the number of attempts at which 75% of students completed the level +1, we're setting a default of 6.5 for the threshold.  This means that levels in CSF courses that do not already have a hint prompt attempts threshold set, will display the hint prompt dialog after 7 runs. 

The number selected for the default is basically arbitrary.  I made it 6.5 so that it would be easily distinguishable from thresholds set programmatically. The distribution of thresholds for the levels shown below was calculated from the data in [thresholds.csv](https://github.com/code-dot-org/code-dot-org/blob/staging/dashboard/scripts/thresholds.csv) which covers levels in Course 2, 3, 4, and A-F 2017.

Screen Shot 2020-08-28 at 3 31 20 PM
 
7 seemed like the middle-ish of the biggest left-hand half and I opted for a lower number than a higher one since showing a dismissible hint prompt when not needed felt like less of a problem than not showing one to a student who was struggling and would benefit from the prompt.  Regardless, a default hint prompt attempts threshold no matter how to you figure it from the levels that currently have hint prompt attempts thresholds set has no real validity for the levels that don't have a hint prompt attempt threshold set because we don't know the relative difficulty of the latter levels nor do we know if they are intended to take more/less attempts to solve. This is a stop-gap solution that gets the hint prompt showing up more frequently than currently, but not optimally. 

# Reviewer Checklist:

- [ ] Tests provide adequate coverage
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: dashboard/app/helpers/levels_helper.rb [hunk 1]
```diff
@@ -349,8 +349,10 @@ def set_tts_options(level_options, app_options)
   end
 
   def set_hint_prompt_options(level_options)
-    if @script && @script.hint_prompt_enabled? && @level.hint_prompt_attempts_threshold
-      level_options[:hintPromptAttemptsThreshold] = @level.hint_prompt_attempts_threshold
+    default_hint_prompt_attempts_threshold = 6.5
+    if @script && @script.hint_prompt_enabled?
+      level_options[:hintPromptAttemptsThreshold] = @level.hint_prompt_attempts_threshold ? @level.hint_prompt_attempts_threshold :
+      default_hint_prompt_attempts_threshold
```
Comment (maddiedierker) [inline dashboard/app/helpers/levels_helper.rb]: nit:

```suggestion
      level_options[:hintPromptAttemptsThreshold] = @level.hint_prompt_attempts_threshold || default_hint_prompt_attempts_threshold
```

File: dashboard/app/helpers/levels_helper.rb [hunk 2]
```diff
@@ -349,8 +349,10 @@ def set_tts_options(level_options, app_options)
   end
 
   def set_hint_prompt_options(level_options)
-    if @script && @script.hint_prompt_enabled? && @level.hint_prompt_attempts_threshold
-      level_options[:hintPromptAttemptsThreshold] = @level.hint_prompt_attempts_threshold
+    default_hint_prompt_attempts_threshold = 6.5
+    if @script && @script.hint_prompt_enabled?
```
Comment (maddiedierker) [inline dashboard/app/helpers/levels_helper.rb]: nit:

```suggestion
    if @script&.hint_prompt_enabled?
```

**Review by maddiedierker — APPROVED** [review]
left some nit comments, but this looks great! thank you for the in-depth description around how you chose this threshold 😄

**Review by Hamms — COMMENTED** [review]
> I made it 6.5 so that it would be easily distinguishable from thresholds set programmatically

I love this! But could we get some comments in the code explaining it? I can easily see how it would be hard for a future maintainer to latch on to the cleverness at work here without some guidance.

**Comment by Erin007** [comment]
> > I made it 6.5 so that it would be easily distinguishable from thresholds set programmatically
> 
> I love this! But could we get some comments in the code explaining it? I can easily see how it would be hard for a future maintainer to latch on to the cleverness at work here without some guidance.

Great idea - I'll add.

**Pull request merged**